### DISCUSSION TITLE
Do not round option when casting to decimal, double, ...

### DISCUSSION THREAD

**Discussion by ErikJansenIRefact** [duckdb/duckdb#14807] [Ideas]
Often data is provided in text format. When decimal values are provided as strings and a given type is expected but not provided automatic rounding prevents proper casting.

An example. Given that an expected column should be of type decimal(4,2) and given that a string value: "9.999" is provided. When applying cast('9.999' as decimal(4,2)) this will result in a value 10.00 instead of throwing an error.

Can a pragma or setting or function argument be provided with which to set the behaviour to not round values when casting and/or try_casting?

**Comment by soerenwolfers** [comment]
Workaround:
```
if(a::T::string=a, a::T, NULL)
```

**Reply by ErikJansenIRefact** [reply]
This does not work.

select '10' as c, if(c::double::string==c, c::double,null) as c_double;

Results in c_double as null. This is caused by casting to double results in 10.0 and when cast back to string results in '10.0' which is different from '10'.

Similar problem given my initial example. Due to rounding the result will change and cast back to string results in different string value.

**Reply by soerenwolfers** [reply]
ah yes. I'm with you, there should be an option for this.
